portal-web/docroot/html/taglib/ui/page_iterator/start.jsp

```diff
@@ -524,6 +524,68 @@ NumberFormat numberFormat = NumberFormat.getNumberInstance(locale);
 	}
 </aui:script>
 
+<aui:script senna="temporary" type="text/javascript">
+	(function () {
+		var pageIterator = document.getElementById('<%= namespace + id %>');
+		if (pageIterator) {
+			var button = pageIterator.querySelector('.dropdown-toggle');
+			var list = pageIterator.querySelector('.dropdown-menu');
+			var options = list.querySelectorAll('.dropdown-item');
+			function onButtonKeyDown(event) {
+				if (
+					event.key === 'ArrowDown' ||
+					event.key === 'ArrowUp' ||
+					event.key === 'Enter' ||
+					event.key === ' '
+				) {
+					event.preventDefault();
+					button.setAttribute('aria-expanded', 'true');
+					list.classList.add('show');
+					if (options) {
+						options[0].focus();
+					}
+				}
+			}
+			button.addEventListener('keydown', onButtonKeyDown);
+			function onLeaveDropdown() {
+				button.setAttribute('aria-expanded', 'false');
+				list.classList.remove('show');
+			}
+			function handleKeyEvents(event) {
+				var currentIndex = Array.from(options).indexOf(
+					document.activeElement
+				);
+				if (event.key === 'ArrowDown') {
+					event.preventDefault();
+					if (currentIndex < options.length - 1) {
+						options[currentIndex + 1].focus();
+					}
+				}
+				else if (event.key === 'ArrowUp') {
+					event.preventDefault();
+					if (currentIndex > 0) {
+						options[currentIndex - 1].focus();
+					}
+				}
+				else if (event.key === 'Escape' || event.key === 'Tab') {
+					button.focus();
+					onLeaveDropdown();
+				}
+				else if (event.key === ' ') {
+					event.preventDefault();
+					options[currentIndex].click();
+				}
+			}
+			list.addEventListener('keydown', handleKeyEvents);
+			var destroyDropDownPagination = function () {
+				button.removeEventListener('keydown', onButtonKeyDown);
+				list.removeEventListener('keydown', handleKeyEvents);
+			};
+			Liferay.once('beforeScreenFlip', destroyDropDownPagination);
+		}
+	})();
+</aui:script>
```

Comment:
out of curiosity, almost the same script happens a bit above in this file, for the delta dropdown.

Just 2 questions:

- Any reason not to handle `focusout` in this script, like in the other?
- Perhaps does not worth it, but, have you considered joining both? they look similar enough except the "source" of  the operation (`pageIterator` here, `dropdown` there)

It's a maintenance trade off as we don't want to spend much time with this but also we might need to change these in the future till we get rid of the code.

I guess you already considered all the above so this is not a blocker.